src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseDeprecatedSecurityProtocols.cs

```diff
@@ -94,31 +131,64 @@ bool IsReferencingDeprecatedProtocol(IFieldReferenceOperation fieldReferenceOper
 
                             // Find the topmost operation with a bad bit set, unless we find an operation that would've been
                             // flagged by the FieldReference callback above.
-                            IOperation foundOperation = null;
+                            IOperation foundDeprecatedOperation = null;
+                            bool foundDeprecatedReference = false;
+                            IOperation foundHardCodedOperation = null;
+                            bool foundHardCodedReference = false;
                             foreach (IOperation childOperation in assignmentOperation.Value.DescendantsAndSelf())
                             {
                                 if (childOperation is IFieldReferenceOperation fieldReferenceOperation
-                                    && IsReferencingDeprecatedProtocol(fieldReferenceOperation))
+                                    && IsReferencingSecurityProtocolType(
+                                        fieldReferenceOperation,
+                                        out var isDeprecatedProtocol,
+                                        out var isHardCodedOkayProtocol))
                                 {
-                                    // This assignment is handled by the FieldReference callback above.
-                                    return;
+                                    if (isDeprecatedProtocol)
+                                    {
+                                        foundDeprecatedReference = true;
+                                    }
+                                    else if (isHardCodedOkayProtocol)
+                                    {
+                                        foundHardCodedReference = true;
+                                    }
+
+                                    if (foundDeprecatedReference && foundHardCodedReference)
```

Comment:
Why return when found both deprecated reference and hardcode reference?
What about this test case:
ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11 | (SecurityProtocolType)192;
Just report diagnostic for Tls12 and Tls11?

Reply:
Because no diagnostics will be reported after the for loop in that case.

For that test case, yep, just report on Tls12 and Tls11.